add GetTimeZonesForRegion

fabriziomoscon opened the pull request:
First attempt...

Retrieve the timezone of a string phone number passed. It uses the same data used by googlei18n/libphonenumber

related to this discussion #30

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ttacon/libphonenumber/31)

<!-- Reviewable:end -->

fabriziomoscon commented:
does this PR need more work @ttacon ?

ttacon commented on countryCodeToTimeZones.go:
```diff
@@ -0,0 +1,2261 @@
+
+import "strconv"
+
+// TODO(ttacon): do we need to worry about load factor?
```
I would been somewhat concerned. Could you test some build times with and w/o this file?

ttacon commented:
Hey! Sorry, just a quick question about load time!

fabriziomoscon commented:
hi,
I have just copied it from this:
https://github.com/ttacon/libphonenumber/blob/5cb77679a4c77d45f2496c9ed8e60b5eec03bb47/countrycodetoregionmap.go#L3
I was not quite sure what the comment meant. Maybe you could remember why did you add the comment in the first place.

fabriziomoscon commented:
@ttacon the build time for this branch is not affected by the new file, so I remove the comment. Please let me know if there anything else to do before merging

ttacon commented on countryCodeToTimeZones.go:
```diff
@@ -0,0 +1,2259 @@
+
+const MAX_REGION_CODE_LENGTH = 7
+
+var CountryCodeToTimeZones = map[int][]string{
```
Can we add a comment here as to what the structure of this mapping means semantically?

ttacon commented:
If you can add a comment to the semantic structure of the `CountyCodeToTimeZones` map, I'll merge this after!

fabriziomoscon commented:
@ttacon I have added this comment:
```
// This structure maps telephone number digits to a particular timezone,
// because a timezones could represent many prefixes, there could be many
// entries for the same timezone.
// For example this is the case for America/New_York.
```
and squashed the 3 commits.

ttacon commented:
Thanks for the contribution!
